paperqa/clients/semantic_scholar.py

```diff
@@ -0,0 +1,310 @@
+        offset: int = 0,
+        limit: int = 1,
+        include_base_url: bool = True,
+    ) -> tuple[str, dict[str, Any]]:
+        """
+        Make the target URL and in-place update the input URL parameters.
+
+        Args:
+            params: URL parameters to in-place update.
+            query: Either a search query or a Semantic Scholar paper ID.
+            offset: Offset to place in the URL parameters for the default search type.
+            limit: Limit to place in the URL parameters for some search types.
+            include_base_url: Set True (default) to include the base URL.
+
+        Returns:
+            Two-tuple of URL and URL parameters.
+        """
+        base = SEMANTIC_SCHOLAR_BASE_URL if include_base_url else ""
+        if self == SematicScholarSearchType.DEFAULT:
+            params["query"] = query.replace("-", " ")
+            params["offset"] = offset
+            params["limit"] = limit
+            return f"{base}/graph/v1/paper/search", params
+        if self == SematicScholarSearchType.PAPER:
+            return f"{base}/graph/v1/paper/{query}", params
+        if self == SematicScholarSearchType.PAPER_RECOMMENDATIONS:
+            return f"{base}/recommendations/v1/papers/forpaper/{query}", params
+        if self == SematicScholarSearchType.DOI:
+            return f"{base}/graph/v1/paper/DOI:{query}", params
+        if self == SematicScholarSearchType.FUTURE_CITATIONS:
+            params["limit"] = limit
+            return f"{base}/graph/v1/paper/{query}/citations", params
+        if self == SematicScholarSearchType.PAST_REFERENCES:
+            params["limit"] = limit
+            return f"{base}/graph/v1/paper/{query}/references", params
+        if self == SematicScholarSearchType.GOOGLE:
+            params["limit"] = 1
+            return f"{base}/graph/v1/paper/search", params
+        if self == SematicScholarSearchType.MATCH:
+            return f"{base}/graph/v1/paper/search/match", params
+        raise NotImplementedError
+
+
+def s2_authors_match(authors: list[str], data: dict):
+    """Check if the authors in the data match the authors in the paper."""
+    AUTHOR_NAME_MIN_LENGTH = 2
+    s2_authors_noinit = [
+        " ".join([w for w in a["name"].split() if len(w) > AUTHOR_NAME_MIN_LENGTH])
+        for a in data["authors"]
+    ]
+    authors_noinit = [
+        " ".join([w for w in a.split() if len(w) > AUTHOR_NAME_MIN_LENGTH])
+        for a in authors
+    ]
+    # Note: we expect the number of authors to be possibly different
+    return any(
+        starmap(
+            lambda x, y: x in y or y in x,
+            zip(s2_authors_noinit, authors_noinit, strict=False),
+        )
+    )
+
+
+async def parse_s2_to_doc_details(
+    paper_data: dict, session: aiohttp.ClientSession
+) -> DocDetails:
+
+    bibtex_source = "self_generated"
+
+    if "data" in paper_data:
+        paper_data = paper_data["data"][0]
+
+    if "ArXiv" in paper_data["externalIds"]:
```

Comment:
Do you mind moving `"ArXiv"` to a module level variable? Or you like it here

Reply:
I don't think this has much need outside of this parsing, so I think a module level variable could be more confusing.